src/services/projectService.js

```diff
@@ -0,0 +1,98 @@
+// import Boom from 'boom';
+import Project from '../models/projects';
+import AdminProject from '../models/admin_project';
+import Admin from '../models/admins';
+
+/**
+ * Get all admin.
+ *
+ * @return {Promise}
+ */
+export function getAllProjects() {
+  return Project.fetchAll();
+}
+
+/**
+ * Get a admin.
+ *
+ * @param  {Number|String}  id
+ * @return {Promise}
+ */
+export async function getRelatedProject(headers) {
+  const email = headers.email;
+  const adminId = await Admin.forge({
+    email: email
+  })
+    .fetch()
+    .then(data => {
+      const pId = data.get('id');
+
+      return pId;
+    });
+
+  const projects = await new AdminProject()
+    .query(function(qb) {
+      qb
+        .where({
+          admin_id: adminId
+        })
+        .select('project_id');
+    })
+    .fetchAll()
+    .then(data => {
+      const result = data.toJSON();
+
+      return result;
+    });
+
+  const projectId = [];
+
+  projects.forEach(element => {
```

Comment:
Use .map function for this purpose

Reply:
Actually, avoid this approach as a whole. Try to filter the projects at database level itself using joins